native/src/main/java/io/ballerina/stdlib/oracledb/parameterprocessor/OracleDBStatementParameterProcessor.java

```diff
@@ -73,6 +75,37 @@ protected void setCustomSqlTypedParam(Connection connection, PreparedStatement p
         }
     }
 
+    @Override
+    public int getCustomOutParameterType(BObject typedValue) throws ApplicationError {
+        String sqlType = typedValue.getType().getName();
+        int sqlTypeValue;
+        switch (sqlType) {
+            case Constants.Types.OutParameterTypes.XML:
+                sqlTypeValue = Types.SQLXML;
+                break;
+            default:
+                throw new ApplicationError("Unsupported OutParameter type: " + sqlType);
+        }
+        return sqlTypeValue;
+    }
+
+    @Override
+    protected void setXml(Connection connection, PreparedStatement preparedStatement,
+                          int index, BXml value) throws SQLException, ApplicationError {
+        if (value == null) {
+            preparedStatement.setNull(index, Types.NULL);
+        } else {
+            try {
+                SQLXML sqlXml = connection.createSQLXML();
+                sqlXml.setString(value.toString());
+                preparedStatement.setObject(index, sqlXml, Types.SQLXML);
+            } catch (NoClassDefFoundError e) {
+                throw new ApplicationError("Error occurred while setting an xml data. Check whether both " +
+                        "`xdb.jar` and `xmlparserv2.jar` are present in the dependency jar list");
```

Comment:
```suggestion
                        "`xdb.jar` and `xmlparserv2.jar` are added as dependency in Ballerina.toml");
```

Reply:
done in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/10/commits/e325a68ee55c9bd5a7b17dbad611333730c549df